### ISSUE TITLE
Jumping errors when there are several adjacent placeholders

### ISSUE THREAD

**Issue by xiuzbl** [SirVer/ultisnips#1359]
For example, I use the following snippets
snippet test "test snippets" iA
$1$1$2(some other things)$0
endsnippet

If I type "testwords", then
**Expected behavior:**
wordswords(some other things)

**Actual behavior:**
wwords(some other things)

The only first letter w repeated and the cursor will jump to location belonging to second
$1 placeholder. 

Actually, if there are any other characters between $1, $1 or $2, the snippets work
as I expected. For example, if I use
snippet test "test snippets" iA
$1$1 $2(some other things)$0
endsnippet
and type "testwords", it will give 
wordswords (some other things)

1.
2.
3.

-----

- **Operating System**: Mac OS 11.3
- **Vim Version**: 
- VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Dec 28 2020 13:00:25)
macOS version
Included patches: 1-2164
- **UltiSnips Version**:3.1 
- **Python inside Vim**: 3.9.1
- **Docker repo/vimrc**:

**Comment by hl037** [comment]
99% chance it's related to #1403 
Placeholders are 0-length text objects, and they are definitively prone to be messed up by the tie breaker. I'm working on this issue and I will soon propose a PR solving I hope many problems with weird jumps

**Comment by SirVer** [comment]
I agree with @hl037, chances are it will improve with the work Léo is currently doing.

**Comment by hl037** [comment]
btw, you may try with #1408  ...But there is low chances it fixes it since it's also related to #1405
